jumanji/environments/packing/bin_pack/generator.py

```diff
@@ -1311,3 +1314,375 @@ def _unpack_items(self, state: State) -> State:
+            std aren't provided.
+        """
+        super().__init__(
+            max_num_items,
+            max_num_ems,
+            True,
+            True,
+            split_eps,
+            prob_split_one_item,
+            split_num_same_items,
+            container_dims,
+            mean_item_value,
+            std_item_value,
+        )
+        self.mean_item_value = mean_item_value
+        self.std_item_value = std_item_value
+        self.min_target_volume = min_target_volume
+        self.max_target_volume = max_target_volume
+        if mean_item_value is None or std_item_value is None:
+            std_name = "std_item_value"
+            raise ValueError(
+                "Value Based generator not provided with"
+                + f"{'mean_item_value attribute' if mean_item_value is None else ''}"
+                + f"{f'{std_name} attribute' if std_item_value is None else ''}"
+            )
+
+    def _generate_value_based_solved_instance(  # type: ignore
+        self, key: chex.PRNGKey, target_volume: int
+    ) -> State:
+        """Generate the random instance with 1/target_volume of all items correctly packed. The
+        other part of the generated items remain unpacked.
+        """
+
+        key, split_key = jax.random.split(key)
+        container = make_container(self.container_dims)
+
+        list_of_ems = [container] + (self.max_num_ems - 1) * [empty_ems()]
+        ems = tree_transpose(list_of_ems)
+        ems_mask = jnp.zeros(self.max_num_ems, bool)
+
+        # Create less than half of max_num_items item spaces by splitting up a container. This
+        # will lead to nb_items_in_one_container items that fit perfectly into a single container.
+        nb_items_in_one_container = math.floor(1 / target_volume * self.max_num_items)
+        # This will generate items_spaces and item_mask of size nb_items_in_one_container.
+        items_spaces, items_mask = self._split_container_into_items_spaces(
+            container, split_key, nb_items_in_one_container
+        )
+        # Randomly generate values that will then be increased by a value of
+        # total_value_of_generated_values to generate a "perfect instance" with a known optimal
+        # solution.
+        key, split_key = jax.random.split(key)
+        item_values = self.mean_item_value + (
+            self.std_item_value
+            * jax.random.normal(split_key, (len(items_mask),), jnp.float32)
+        )
+        # Assign values to the items that are packed in the optimal solution. To ensure an optimal
+        # solution, the total value of the duplicated item values are added to the original
+        # generated item values.
+        optimal_items = valued_item_from_space_and_max_value(items_spaces, item_values)
+        # Duplicate the above items and assign values to them that are half their counterparts
+        # above.
+        extra_items = valued_item_from_space_and_max_value(items_spaces, item_values)
```

Comment:
Given that we aren't forcing the ability to get an optimal value here anymore you can generate item_values over the total length of items and assign them directly over the whole space.